### ISSUE TITLE
NetworkInformation tests show weird error on macOS

### ISSUE THREAD

**Issue by wfurt** [dotnet/runtime#64441]
```
===========================================================================================================
  ~/github/wfurt-ssl/artifacts/bin/System.Net.NetworkInformation.Functional.Tests/net7.0-Debug ~/github/wfurt-ssl/src/libraries/System.Net.NetworkInformation/tests/FunctionalTests
    Discovering: System.Net.NetworkInformation.Functional.Tests (method display = ClassAndMethod, method display options = None)
  ps: illegal option -- -
  usage: ps [-AaCcEefhjlMmrSTvwXx] [-O fmt | -o fmt] [-G gid[,gid...]]
            [-g grp[,grp...]] [-u [uid,uid...]]
            [-p pid[,pid...]] [-t tty[,tty...]] [-U user[,user...]]
         ps [-L]
    Discovered:  System.Net.NetworkInformation.Functional.Tests (found 76 of 98 test cases)
    Starting:    System.Net.NetworkInformation.Functional.Tests (parallel test collections = on, max threads = 16)
    Finished:    System.Net.NetworkInformation.Functional.Tests
  === TEST EXECUTION SUMMARY ===
     System.Net.NetworkInformation.Functional.Tests  Total: 242, Errors: 0, Failed: 0, Skipped: 0, Time: 0.593s
```

seems like regression from #63788 assuming gnu `ps`. Ignoring busybox `ps` and other OSes.
probably not critical but nice to fix. 

cc: @ManickaP @tmds

**Comment by tmds** [comment]
I'll look into this as part of https://github.com/dotnet/runtime/issues/64319 early next week.

**Comment by ManickaP** [comment]
Yeah we need to redirect std err as well. But this is just making mess on the output, not affecting the actual tests.

**Comment by wfurt** [comment]
note that macOS _can_ print threads as well but the syntax is different. This should probably go via the abstraction/wrapper we have for Ping itself. (or we should find better way how to dump threads. If we make the changes @tmds suggested and we use ReadAsync, the dedicated thread would vanish completely.

**Comment by karelz** [comment]
This should be addressed in PR #64614 (deletes the test).

**Comment by wfurt** [comment]
looks good. test run is clean.
